lib/kubernetes-deploy/template_sets.rb

```diff
@@ -106,18 +109,26 @@ def from_dirs_and_files(paths:, logger:)
       end
     end
 
-    def with_resource_definitions(render_erb: false, current_sha: nil, bindings: nil)
-      @template_sets.each do |template_set|
-        template_set.with_resource_definitions(
+    def with_resource_definitions_and_filename(render_erb: false, current_sha: nil, bindings: nil, raw: false)
+      with_delayed_exceptions(@template_sets, KubernetesDeploy::InvalidTemplateError) do |template_set|
+        template_set.with_resource_definitions_and_filename(
           render_erb: render_erb,
           current_sha: current_sha,
-          bindings: bindings
-        ) do |r_def|
-          yield r_def
+          bindings: bindings,
+          raw: raw
+        ) do |r_def, filename|
+          yield r_def, filename
         end
       end
     end
 
+    def with_resource_definitions(render_erb: false, current_sha: nil, bindings: nil, raw: false)
```

Comment:
Open question should I remove this function and just change everything using it to actually use `with_resource_definitions_and_filename `

Reply:
I'm either/or on this. I think the clarity it brings to the code (e.g. not needing to care about `filename` in certain cases) makes it worthwhile. Since it's really just a wrapper around `with_resource_definitions_and_filename` it's not really duplicating _too much_.